<?xml version="1.0"?>
<launch>

	<arg name="name_tag" default="tag_0" />	<!-- Change also in marker_detector.launch -->
	<arg name="name_file" default="tag_detection_pose.txt" />
	<arg name="topic_tag" default="/camera_network/tag_detections_pose" /> <!-- Same in marker_detector.launch -->
	<arg name="output_topic" default="arcore/origin" />
	<arg name="tracker_topic" default="/tracker/tracks_smoothed" />
	<arg name="tracker_arcore" default="arcore/centroid_mod" />
	<arg name="sensor_name" default="kinect2_head" />

	<!-- Apriltag detector -->
	<include file="$(find arcore)/launch/marker_detector.launch">
		<arg name="sensor_name" value="$(arg sensor_name)" />
	</include>


	<!-- Server node for mobile comunication -->
	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch">
		<param name="port" value="9090"/>
	</include>

	
	<!-- Mobile node publisher -->
	<node name="mobile_receiver" pkg="arcore" type="mobile_receiver" output="screen">
		<param name="input_topic" value="/arcore/vodom"/>
		<param name="output_topic" value="/arcore/marker_array"/>
		<param name="name_tag" value="$(arg name_tag)_arcore"/>
		<param name="origin" value="$(arg output_topic)"/>
	</node>
	
	<!-- Keep Tag Living  -->
	<node name="keep_tag_living" pkg="arcore" type="keep_tag_living" output="screen">
		<param name="file_path" value="$(find arcore)/file/$(arg name_file)"/>
		<param name="output_name_tag" value="$(arg name_tag)_arcore"/>
	</node>


	<!-- Tag File -->
	<node name="tag_file" pkg="arcore" type="tag_file" output="screen">
		<param name="file_path" value="$(find arcore)/file/$(arg name_file)"/>
		<param name="name_tag" value="$(arg name_tag)"/>
		<param name="input_topic" value="$(arg topic_tag)"/>
	</node>
	
	<!-- Origin sender -->
	<node name="origin_sender" pkg="arcore" type="origin_sender" output="screen">
		<param name="name_tag" value="$(arg name_tag)_arcore"/>
		<param name="output_topic" value="$(arg output_topic)"/>
	</node>
	
	<!-- Modifier Track -->
	<node name="centroid_modifier" pkg="arcore" type="centroid_modifier" output="screen">
		<param name="input_track" value="$(arg tracker_topic)"/>
		<param name="origin" value="$(arg output_topic)"/>
		<param name="output_topic" value="$(arg tracker_arcore)"/>
	</node>
  
</launch>
